Code/CryScriptSystem/ScriptBindings/ScriptBind_System.cpp

```diff
@@ -1823,4 +1827,46 @@ int ScriptBind_System::GetEntityScriptFilePath(IFunctionHandler* pH, const char*
 		return pH->EndFunction();
 	}
 	return pH->EndFunction(pClass->GetScriptFile());
+}
+
+int ScriptBind_System::GetItemClasses(IFunctionHandler * pH)
+{
+	SmartScriptTable pObj(m_pSS);
+	int k = 1;
+
+	IItemSystem* pItemSystem = gEnv->pGame->GetIGameFramework()->GetIItemSystem();
+	IEntityClassRegistry* pEntityRegistry = gEnv->pEntitySystem->GetClassRegistry();
+	IEntityClass* pClass;
+
+	for (pEntityRegistry->IteratorMoveFirst(); pClass = pEntityRegistry->IteratorNext();)
+	{
+		if (pClass) {
+			if (pItemSystem->IsItemClass(pClass->GetName()))
+				pObj->SetAt(k++, pClass->GetName());
+
+		}
+	}
+
+	return pH->EndFunction(*pObj);
+}
+
+int ScriptBind_System::GetVehicleClasses(IFunctionHandler * pH)
+{
+	SmartScriptTable pObj(m_pSS);
+	int k = 1;
+
+	IVehicleSystem* pVehicleSystem = gEnv->pGame->GetIGameFramework()->GetIVehicleSystem();
+	IEntityClassRegistry* pEntityRegistry = gEnv->pEntitySystem->GetClassRegistry();
+	IEntityClass* pClass;
+
+	for (pEntityRegistry->IteratorMoveFirst(); pClass = pEntityRegistry->IteratorNext();)
+	{
+		if (pClass) {
+			if (pVehicleSystem->IsVehicleClass(pClass->GetName()))
+				pObj->SetAt(k++, pClass->GetName());
+
+		}
```

Comment:
```suggestion
		if (pVehicleSystem->IsVehicleClass(pClass->GetName()))
		{
			pObj->SetAt(k++, pClass->GetName());
		}
```

When `pClass` is null, the end has been reached, so the loop body is not called again.